apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx

```diff
@@ -186,47 +194,168 @@ export const getServerSideProps: GetServerSideProps<PageProps> = async ({
 
 export const PageNotificationBanner = ({
   archived = false,
+  pageMetadata,
+  onUnarchived,
 }: {
   archived: boolean;
+  pageMetadata: EntityMetadata;
+  onUnarchived: () => void;
 }) => {
   const { pageEntityId } = usePageContext();
-  const [archivePage] = useArchivePage();
+  const { unarchivePage } = useArchivePage();
+
+  const { provenance, temporalVersioning } = pageMetadata;
+
+  const { users } = useUsers();
+
+  const archivedByAccountId = provenance.recordCreatedById;
+
+  const archivedByUser =
+    users && users.find(({ accountId }) => archivedByAccountId === accountId);
+
+  const archivedAt = useMemo(
+    () => new Date(temporalVersioning.decisionTime.start.limit),
+    [temporalVersioning],
+  );
+
+  const timeSinceArchived = useMemo(() => {
+    const now = new Date();
+
+    const differenceInMilliseconds = now.getTime() - archivedAt.getTime();
+
+    const differenceInDays = Math.round(
+      differenceInMilliseconds / (1000 * 60 * 60 * 24),
+    );
+
+    if (differenceInDays >= 1) {
+      return `${differenceInDays} days`;
+    } else {
+      const differenceInHours = Math.round(
+        differenceInMilliseconds / (1000 * 60 * 60),
+      );
+      if (differenceInHours >= 1) {
+        return `${differenceInHours} hours`;
+      } else {
+        const differenceInMinutes = Math.round(
+          differenceInMilliseconds / (1000 * 60),
+        );
+
+        return `${differenceInMinutes} minutes`;
+      }
+    }
+  }, [archivedAt]);
```

Comment:
We have `date-fns` in the repo [[docs](https://date-fns.org/v2.30.0/docs/formatDistance)]
```suggestion
  const timeSinceArchived = formatDistance(archivedAt, new Date());
```